diff -urN docky-2.2.0/Docky.Services/Docky.Services/SystemService.cs docky-2.2.0-patched/Docky.Services/Docky.Services/SystemService.cs
--- docky-2.2.0/Docky.Services/Docky.Services/SystemService.cs	2013-01-10 08:07:47.000000000 -0500
+++ docky-2.2.0-patched/Docky.Services/Docky.Services/SystemService.cs	2014-11-02 11:25:02.228821566 -0500
@@ -278,7 +278,7 @@
 			event Action Changed;
 		}
 
-		[Interface(UPowerName)]
+		[Interface("org.freedesktop.DBus.Properties")]
 		interface IUPower : org.freedesktop.DBus.Properties
 		{
 			event Action Changed;
@@ -338,11 +338,16 @@
 
 		void HandleUPowerChanged ()
 		{
-			bool newState = (bool) upower.Get (UPowerName, "OnBattery");
-			
-			if (on_battery != newState) {
-				on_battery = newState;
-				OnBatteryStateChanged ();
+			try {
+				bool newState = (bool) upower.Get (UPowerName, "OnBattery");
+				
+				if (on_battery != newState) {
+					on_battery = newState;
+					OnBatteryStateChanged ();
+				}
+			} catch (Exception e) {
+				Log<SystemService>.Error ("Could not initialize power manager dbus: '{0}'", e.Message);
+				Log<SystemService>.Info (e.StackTrace);
 			}
 		}
 
