diff '--exclude=.git' -urN a/meson.build b/meson.build
--- a/meson.build	2019-05-02 08:56:52.000000000 -0500
+++ b/meson.build	2019-12-22 22:11:17.614486452 -0600
@@ -66,10 +66,31 @@
 use_launcher = get_option('launcher')
 if use_launcher
         dep_expat = dependency('expat', version: '>=2.2')
-        dep_libsystemd = dependency('libsystemd', version: '>=230')
-        dep_systemd = dependency('systemd', version: '>=230')
 
-        add_project_arguments('-DSYSTEMUIDMAX=' + dep_systemd.get_pkgconfig_variable('systemuidmax'), language: 'c')
+        enable_elogind = get_option('elogind')
+        if enable_elogind
+            dep_libsystemd = dependency('libelogind', version: '>= 230')
+            dep_systemd = dependency('libelogind', version: '>=230')
+
+            awk = find_program('awk')
+            system_uid_max = run_command(
+                    awk,
+                    '/^\s*SYS_UID_MAX\s+/ { uid=$2 } END { print uid }',
+                    '/etc/login.defs').stdout().strip()
+            if system_uid_max == ''
+                    system_uid_max = '999'
+            endif
+            system_uid_max = system_uid_max.to_int()
+
+            add_project_arguments('-DSYSTEMUIDMAX=' + system_uid_max.to_string(), language: 'c')
+        else
+            dep_libsystemd = dependency('libsystemd', version: '>=230')
+            dep_systemd = dependency('systemd', version: '>=230')
+
+            add_project_arguments('-DSYSTEMUIDMAX=' + dep_systemd.get_pkgconfig_variable('systemuidmax'), language: 'c')
+        endif
+        conf.set('HAVE_ELOGIND', enable_elogind)
+
         conf.set('systemunitdir', dep_systemd.get_pkgconfig_variable('systemdsystemunitdir'))
         conf.set('userunitdir', dep_systemd.get_pkgconfig_variable('systemduserunitdir'))
         conf.set('catalogdir', dep_systemd.get_pkgconfig_variable('catalogdir'))
diff '--exclude=.git' -urN a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2019-05-02 08:56:52.000000000 -0500
+++ b/meson_options.txt	2019-12-22 22:04:29.116660918 -0600
@@ -1,6 +1,7 @@
 option('audit', type: 'boolean', value: false, description: 'Audit support')
 option('launcher', type: 'boolean', value: true, description: 'Build compatibility launcher')
 option('docs', type: 'boolean', value: false, description: 'Build documentation')
+option('elogind', type: 'boolean', value: true, description: 'Use elogind')
 option('reference-test', type: 'boolean', value: false, description: 'Run test suite against reference implementation')
 option('selinux', type: 'boolean', value: false, description: 'SELinux support')
 option('system-console-users', type: 'array', value: [], description: 'Additional set of names of system-users to be considered at-console')
diff '--exclude=.git' -urN a/src/launch/main.c b/src/launch/main.c
--- a/src/launch/main.c	2019-05-02 08:56:52.000000000 -0500
+++ b/src/launch/main.c	2019-12-22 22:06:26.373035232 -0600
@@ -6,7 +6,13 @@
 #include <fcntl.h>
 #include <getopt.h>
 #include <stdlib.h>
+
+#ifdef HAVE_ELOGIND
+#include <elogind/sd-daemon.h>
+#else
 #include <systemd/sd-daemon.h>
+#endif
+
 #include "launch/launcher.h"
 #include "util/error.h"
 
diff '--exclude=.git' -urN a/src/launch/policy.c b/src/launch/policy.c
--- a/src/launch/policy.c	2019-05-02 08:56:52.000000000 -0500
+++ b/src/launch/policy.c	2019-12-22 22:04:29.116660918 -0600
@@ -6,7 +6,13 @@
 #include <c-rbtree.h>
 #include <c-stdaux.h>
 #include <stdlib.h>
+
+#ifdef HAVE_ELOGIND
+#include <elogind/sd-bus.h>
+#else
 #include <systemd/sd-bus.h>
+#endif
+
 #include "dbus/protocol.h"
 #include "launch/config.h"
 #include "launch/policy.h"
diff '--exclude=.git' -urN a/src/launch/policy.h b/src/launch/policy.h
--- a/src/launch/policy.h	2019-05-02 08:56:52.000000000 -0500
+++ b/src/launch/policy.h	2019-12-22 22:04:29.116660918 -0600
@@ -8,7 +8,13 @@
 #include <c-rbtree.h>
 #include <c-stdaux.h>
 #include <stdlib.h>
+
+#ifdef HAVE_ELOGIND
+#include <elogind/sd-bus.h>
+#else
 #include <systemd/sd-bus.h>
+#endif
+
 #include "launch/config.h"
 
 typedef struct Policy Policy;
diff '--exclude=.git' -urN a/test/dbus/util-broker.c b/test/dbus/util-broker.c
--- a/test/dbus/util-broker.c	2019-05-02 08:56:52.000000000 -0500
+++ b/test/dbus/util-broker.c	2019-12-22 22:04:29.116660918 -0600
@@ -12,8 +12,15 @@
 #include <sys/types.h>
 #include <sys/un.h>
 #include <sys/wait.h>
+
+#ifdef HAVE_ELOGIND
+#include <elogind/sd-bus.h>
+#include <elogind/sd-event.h>
+#else
 #include <systemd/sd-bus.h>
 #include <systemd/sd-event.h>
+#endif
+
 #include "dbus/protocol.h"
 #include "util/syscall.h"
 #include "util-broker.h"
diff '--exclude=.git' -urN a/test/dbus/util-broker.h b/test/dbus/util-broker.h
--- a/test/dbus/util-broker.h	2019-05-02 08:56:52.000000000 -0500
+++ b/test/dbus/util-broker.h	2019-12-22 22:04:29.116660918 -0600
@@ -11,8 +11,14 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/un.h>
+
+#ifdef HAVE_ELOGIND
+#include <elogind/sd-bus.h>
+#include <elogind/sd-event.h>
+#else
 #include <systemd/sd-bus.h>
 #include <systemd/sd-event.h>
+#endif
 
 typedef struct Broker Broker;
 
