diff --git a/CMakeLists.txt b/CMakeLists.txt
index d5aad80..47916e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -258,7 +258,7 @@ endif()
 set(VERSION "${PROJECT_VERSION}")
 set(prefix "${CMAKE_INSTALL_PREFIX}")
 set(exec_prefix "\${prefix}")
-set(libdir "\${exec_prefix}/lib")
+set(libdir "${LIB_INSTALL_DIR}")
 set(includedir "\${prefix}/include")
 set(PTHREAD_LIBS "${CMAKE_THREAD_LIBS_INIT}")
 
diff --git a/libical.pc.in b/libical.pc.in
index 6ca4c7c..7c1c536 100644
--- a/libical.pc.in
+++ b/libical.pc.in
@@ -7,5 +7,5 @@ threadslib=@PTHREAD_LIBS@
 Name: libical
 Description: An implementation of basic iCAL protocols
 Version: @VERSION@
-Libs: -L${libdir} -lical -licalss -licalvcal ${threadslib}
+Libs: -lical -licalss -licalvcal ${threadslib}
 Cflags: -I${includedir}
