diff '--exclude=.git' -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2018-09-18 08:15:11.000000000 -0500
+++ b/Makefile.am	2018-09-23 17:21:39.299381843 -0500
@@ -1026,9 +1026,7 @@
 if HAVE_INTROSPECTION
 
 libnm_noinst_data = \
-	libnm/nm-property-docs.xml \
 	libnm/nm-settings-docs-overrides.xml \
-	libnm/nm-settings-docs.xml \
 	libnm/nm-settings-keyfile-docs.xml \
 	libnm/nm-settings-ifcfg-rh-docs.xml
 
@@ -3468,27 +3466,11 @@
 $(clients_common_libnmc_base_la_OBJECTS): clients/common/.dirstamp
 
 clients_common_settings_doc_h = clients/common/settings-docs.h
-if HAVE_INTROSPECTION
-$(clients_common_settings_doc_h): clients/common/settings-docs.xsl libnm/nm-property-docs.xml clients/common/.dirstamp
-	$(AM_V_GEN) $(XSLTPROC) --output $@ $< $(word 2,$^)
-DISTCLEANFILES += $(clients_common_settings_doc_h)
-check-local-settings-docs: $(clients_common_settings_doc_h)
-	@if test -z "$$NMTST_NO_CHECK_SETTINGS_DOCS" ; then \
-		if ! cmp -s "$(srcdir)/$(clients_common_settings_doc_h).in" "$(builddir)/$(clients_common_settings_doc_h)" ; then \
-			if test "$$NM_TEST_REGENERATE" == 1 ; then \
-				cp -f "$(builddir)/$(clients_common_settings_doc_h)" "$(srcdir)/$(clients_common_settings_doc_h).in"; \
-			else \
-				echo "The generated file \"$(builddir)/$(clients_common_settings_doc_h)\" differs from the source file \"$(srcdir)/$(clients_common_settings_doc_h).in\". You probably should copy the generated file over to the source file. You can skip this test by setting \$$NMTST_NO_CHECK_SETTINGS_DOCS=yes". You can also automatically copy the file by rerunning the test with \$$NM_TEST_REGENERATE=1 ; \
-				false; \
-			fi; \
-		fi;\
-	fi
-check_local += check-local-settings-docs
-else
+
 $(clients_common_settings_doc_h): $(clients_common_settings_doc_h).in clients/common/.dirstamp
 	$(AM_V_GEN) cp "$(srcdir)/$(clients_common_settings_doc_h).in" "$(builddir)/$(clients_common_settings_doc_h)"
 check-local-settings-docs:
-endif
+
 EXTRA_DIST += \
 	$(clients_common_settings_doc_h) \
 	$(clients_common_settings_doc_h).in
diff '--exclude=.git' -urN a/configure.ac b/configure.ac
--- a/configure.ac	2018-09-18 08:15:11.000000000 -0500
+++ b/configure.ac	2018-09-23 17:20:07.738816404 -0500
@@ -1223,10 +1223,6 @@
 # check if we can build setting property documentation
 build_docs=no
 if test -n "$INTROSPECTION_MAKEFILE"; then
-	# If g-i is installed we know we have python, but we might not have pygobject
-	if ! "$PYTHON" -c 'from gi.repository import GObject' >& /dev/null; then
-		AC_MSG_ERROR(["--enable-introspection aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
-	fi
 
 	AC_PATH_PROG(PERL, perl)
 	if test -z "$PERL"; then
