diff '--exclude=.git' -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2018-06-16 02:17:40.000000000 -0500
+++ b/Makefile.am	2018-09-30 17:18:07.125115372 -0500
@@ -916,9 +916,7 @@
 if HAVE_INTROSPECTION
 
 libnm_noinst_data = \
-	libnm/nm-property-docs.xml \
 	libnm/nm-settings-docs-overrides.xml \
-	libnm/nm-settings-docs.xml \
 	libnm/nm-settings-keyfile-docs.xml \
 	libnm/nm-settings-ifcfg-rh-docs.xml
 
@@ -3258,17 +3256,12 @@
 
 
 clients_cli_settings_doc_c = clients/cli/settings-docs.c
-if HAVE_INTROSPECTION
-$(clients_cli_settings_doc_c): clients/cli/settings-docs.xsl libnm/nm-property-docs.xml
-	$(AM_V_GEN) $(XSLTPROC) --output $@ $< $(word 2,$^)
-$(clients_cli_settings_doc_c): clients/cli/.dirstamp
-DISTCLEANFILES += $(clients_cli_settings_doc_c)
-else
+
 $(clients_cli_settings_doc_c):
 	@echo "to generate $(clients_cli_settings_doc_c), configure with --enable-introspection"
 	@echo "alternatively, build --without-nmcli"
 	@false
-endif
+
 EXTRA_DIST += $(clients_cli_settings_doc_c)
 
 
diff '--exclude=.git' -urN a/configure.ac b/configure.ac
--- a/configure.ac	2018-06-16 02:17:40.000000000 -0500
+++ b/configure.ac	2018-09-30 17:12:02.696691491 -0500
@@ -1140,11 +1140,6 @@
 # check if we can build setting property documentation
 build_docs=no
 if test -n "$INTROSPECTION_MAKEFILE"; then
-    # If g-i is installed we know we have python, but we might not have pygobject
-    if ! "$PYTHON" -c 'from gi.repository import GObject' >& /dev/null; then
-        AC_MSG_ERROR(["--enable-introspection aims to build the settings documentation. This requires GObject introspection for python (pygobject)])
-    fi
-
     AC_PATH_PROG(PERL, perl)
     if test -z "$PERL"; then
         AC_MSG_ERROR([--enable-introspection requires perl])
