From 9fd29449575aff54ae0fb57cc8136df8bdbf49af Mon Sep 17 00:00:00 2001
From: pbgc <pbgc@oninet.pt>
Date: Tue, 10 May 2016 05:44:30 +1000
Subject: vc: Add darcs to list of loaded plugins (bgo#766185)

---
 meld/vc/__init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meld/vc/__init__.py b/meld/vc/__init__.py
index b710cb4..656167f 100644
--- a/meld/vc/__init__.py
+++ b/meld/vc/__init__.py
@@ -28,7 +28,7 @@ from . import _null
 
 # FIXME: This is a horrible hack to help cx_Freeze pick up these plugins when
 # freezing the distributable package.
-from . import git, mercurial, bzr, svn
+from . import git, mercurial, bzr, svn, darcs
 
 # Tuple with module name and vc.NAME field, ordered according to best-guess
 # as to which VC a user is likely to want by default in a multiple-VC situation
@@ -37,6 +37,7 @@ vc_names = (
     "mercurial",
     "bzr",
     "svn",
+    "darcs",
 )
 
 _plugins = [importlib.import_module("." + vc, __package__) for vc in vc_names]
-- 
cgit v0.12

