diff -ru dconf-0.38.0.old/client/meson.build dconf-0.38.0/client/meson.build
--- dconf-0.38.0.old/client/meson.build	2020-09-13 04:10:05.902783900 +0300
+++ dconf-0.38.0/client/meson.build	2020-09-14 01:37:52.115228719 +0300
@@ -57,17 +57,21 @@
 )
 
 if get_option('vapi')
-  vapigen_dep = dependency('vapigen')
+  vapigen_dep = dependency('vapigen', required: false)
 
   vapi_data = files(
     'dconf.vapi',
     'dconf.deps',
   )
 
+if vapigen_dep.found()
   vapi_dir = vapigen_dep.get_pkgconfig_variable(
     'vapidir',
     define_variable: ['datadir', dconf_datadir]
   )
+else
+  vapi_dir = join_paths(dconf_datadir, 'vala', 'vapi')
+endif
 
   install_data(
     vapi_data,
