diff '--exclude=.git' -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2019-05-07 05:23:39.000000000 -0500
+++ b/CMakeLists.txt	2019-05-30 00:42:50.649599392 -0500
@@ -25,11 +25,13 @@
     PURPOSE "Required for screencast portal"
 )
 
-find_package(PipeWire)
-set_package_properties(PipeWire PROPERTIES
-    TYPE OPTIONAL
-    PURPOSE "Required for screencast portal"
+if (USE_PIPEWIRE)
+    find_package(PipeWire)
+    set_package_properties(PipeWire PROPERTIES
+        TYPE OPTIONAL
+        PURPOSE "Required for screencast portal"
 )
+endif ()
 
 find_package(GBM)
 set_package_properties(GBM PROPERTIES
