From 0813fcabf5b13b2b90780ec45f3018ad00927da5 Mon Sep 17 00:00:00 2001
From: Behdad Esfahbod <behdad@behdad.org>
Date: Mon, 20 Nov 2017 15:27:48 -0800
Subject: Fix test build

---
 tests/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index d036088..b382e52 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -61,7 +61,7 @@ endif
 
 TEST_PANGO_LIBS = ../pango/libpango-$(PANGO_API_VERSION).la
 TEST_PANGOFT2_LIBS = $(TEST_PANGO_LIBS) ../pango/libpangoft2-$(PANGO_API_VERSION).la
-TEST_PANGOCAIRO_LIBS = $(TEST_PANGO_LIBS) ../pango/libpangocairo-$(PANGO_API_VERSION).la
+TEST_PANGOCAIRO_LIBS = $(TEST_PANGOFT2_LIBS) ../pango/libpangocairo-$(PANGO_API_VERSION).la
 
 test_layout_SOURCES = \
 	test-layout.c \
-- 
cgit v0.12

