diff '--exclude=.git' -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2019-03-31 12:20:47.000000000 -0500
+++ b/Makefile.am	2019-04-08 19:27:24.702957896 -0500
@@ -7,12 +7,6 @@
 	vala \
 	ccode \
 	codegen \
-	compiler \
-	vapi \
-	tests \
-	doc \
-	gobject-introspection \
-	vapigen \
 	$(NULL)
 
 if ENABLE_VALADOC
diff '--exclude=.git' -urN a/configure.ac b/configure.ac
--- a/configure.ac	2019-04-08 13:55:34.000000000 -0500
+++ b/configure.ac	2019-04-08 19:26:27.509211358 -0500
@@ -218,19 +218,6 @@
            ccode/Makefile
            vala/Makefile
            codegen/Makefile
-           compiler/Makefile
-           vapi/Makefile
-           tests/Makefile
-           tests/girwriter/Makefile
-           tests/gtktemplate/Makefile
-           doc/Makefile
-           doc/manual/Makefile
-           doc/manual/version.xml
-           gobject-introspection/Makefile
-           vapigen/vapigen.pc
-           vapigen/Makefile
-           vapigen/vala-gen-introspect/Makefile
-           vapigen/vala-gen-introspect/vala-gen-introspect
            libvaladoc/valadoc.pc
            libvaladoc/valadoc.deps
            libvaladoc/Makefile
