From 0c15e9cd567366bdc50c91a2f3c92d5af52e3195 Mon Sep 17 00:00:00 2001
From: Philip Withnall <withnall@endlessm.com>
Date: Mon, 11 Sep 2017 19:24:06 +0100
Subject: gversionmacros: Add version macros for GLib 2.56

Signed-off-by: Philip Withnall <withnall@endlessm.com>
---
 glib/gversionmacros.h | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/glib/gversionmacros.h b/glib/gversionmacros.h
index c23c40247..df6585c2f 100644
--- a/glib/gversionmacros.h
+++ b/glib/gversionmacros.h
@@ -185,6 +185,16 @@
  */
 #define GLIB_VERSION_2_54       (G_ENCODE_VERSION (2, 54))
 
+/**
+ * GLIB_VERSION_2_56:
+ *
+ * A macro that evaluates to the 2.56 version of GLib, in a format
+ * that can be used by the C pre-processor.
+ *
+ * Since: 2.56
+ */
+#define GLIB_VERSION_2_56       (G_ENCODE_VERSION (2, 56))
+
 /* evaluates to the current stable version; for development cycles,
  * this means the next stable target
  */
@@ -486,4 +496,18 @@
 # define GLIB_AVAILABLE_IN_2_54                 _GLIB_EXTERN
 #endif
 
+#if GLIB_VERSION_MIN_REQUIRED >= GLIB_VERSION_2_56
+# define GLIB_DEPRECATED_IN_2_56                GLIB_DEPRECATED
+# define GLIB_DEPRECATED_IN_2_56_FOR(f)         GLIB_DEPRECATED_FOR(f)
+#else
+# define GLIB_DEPRECATED_IN_2_56                _GLIB_EXTERN
+# define GLIB_DEPRECATED_IN_2_56_FOR(f)         _GLIB_EXTERN
+#endif
+
+#if GLIB_VERSION_MAX_ALLOWED < GLIB_VERSION_2_56
+# define GLIB_AVAILABLE_IN_2_56                 GLIB_UNAVAILABLE(2, 56)
+#else
+# define GLIB_AVAILABLE_IN_2_56                 _GLIB_EXTERN
+#endif
+
 #endif /*  __G_VERSION_MACROS_H__ */
-- 
cgit v1.2.3

