diff '--exclude=.git' -urN a/configure.ac b/configure.ac
--- a/configure.ac	2017-10-02 11:01:23.000000000 -0400
+++ b/configure.ac	2017-10-09 16:05:04.853228496 -0400
@@ -382,14 +382,14 @@
 # option to specify python interpreter to use; this just sets $PYTHON, so that
 # we will fallback to reading $PYTHON if --with-python is not given, and
 # python.m4 will get the expected input
-AC_ARG_WITH(python,
-            AS_HELP_STRING([--with-python=PATH],
-                           [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
-            [PYTHON="$withval"], [])
-if test x"$PYTHON" = xyes; then
-  AC_MSG_ERROR([--with-python option requires a path or program argument])
-fi
-AM_PATH_PYTHON(2.7,,PYTHON="python2.7")
+# AC_ARG_WITH(python,
+#             AS_HELP_STRING([--with-python=PATH],
+#                            [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
+#             [PYTHON="$withval"], [])
+# if test x"$PYTHON" = xyes; then
+#   AC_MSG_ERROR([--with-python option requires a path or program argument])
+# fi
+# AM_PATH_PYTHON(2.7,,PYTHON="python2.7")
 
 
 dnl ***********************
@@ -3469,7 +3469,6 @@
 gobject/tests/Makefile
 gthread/Makefile
 gio/Makefile
-gio/gdbus-2.0/codegen/Makefile
 gio/gdbus-2.0/codegen/config.py
 gio/gnetworking.h
 gio/xdgmime/Makefile
diff '--exclude=.git' -urN a/configure.ac.rej b/configure.ac.rej
--- a/configure.ac.rej	1969-12-31 19:00:00.000000000 -0500
+++ b/configure.ac.rej	2017-10-09 16:01:27.499410763 -0400
@@ -0,0 +1,25 @@
+--- configure.ac
++++ configure.ac
+@@ -385,14 +385,14 @@ AC_SUBST(PERL_PATH)
+ # option to specify python interpreter to use; this just sets $PYTHON, so that
+ # we will fallback to reading $PYTHON if --with-python is not given, and
+ # python.m4 will get the expected input
+-AC_ARG_WITH(python,
+-            AS_HELP_STRING([--with-python=PATH],
+-                           [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
+-            [PYTHON="$withval"], [])
+-if test x"$PYTHON" = xyes; then
+-  AC_MSG_ERROR([--with-python option requires a path or program argument])
+-fi
+-AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
++# AC_ARG_WITH(python,
++#             AS_HELP_STRING([--with-python=PATH],
++#                            [Path to Python interpreter; searches $PATH if only a program name is given; if not given, searches for a few standard names such as "python3" or "python2"]),
++#             [PYTHON="$withval"], [])
++# if test x"$PYTHON" = xyes; then
++#   AC_MSG_ERROR([--with-python option requires a path or program argument])
++# fi
++# AM_PATH_PYTHON(2.5,,PYTHON="/usr/bin/env python2.5")
+ 
+ 
+ dnl ***********************
diff '--exclude=.git' -urN a/docs/reference/gio/Makefile.am b/docs/reference/gio/Makefile.am
--- a/docs/reference/gio/Makefile.am	2016-10-22 01:21:02.000000000 -0400
+++ b/docs/reference/gio/Makefile.am	2017-10-09 16:01:27.500410757 -0400
@@ -177,7 +177,6 @@
 	gsettings.1			\
 	gresource.1			\
 	gdbus.1				\
-	gdbus-codegen.1			\
 	gio.1				\
 	$(NULL)
 
diff '--exclude=.git' -urN a/gio/Makefile.am b/gio/Makefile.am
--- a/gio/Makefile.am	2017-07-13 19:03:38.000000000 -0400
+++ b/gio/Makefile.am	2017-10-09 16:01:27.500410757 -0400
@@ -1,6 +1,6 @@
 include $(top_srcdir)/glib.mk
 
-SUBDIRS = gdbus-2.0/codegen
+SUBDIRS =
 
 if OS_UNIX
 if !OS_COCOA
diff '--exclude=.git' -urN a/gio/tests/Makefile.am b/gio/tests/Makefile.am
--- a/gio/tests/Makefile.am	2017-10-02 10:58:06.000000000 -0400
+++ b/gio/tests/Makefile.am	2017-10-09 16:01:27.500410757 -0400
@@ -460,10 +460,8 @@
 
 BUILT_SOURCES += gdbus-test-codegen-generated.c gdbus-test-codegen-generated.h
 gdbus-test-codegen.o: gdbus-test-codegen-generated.h
-gdbus-test-codegen-generated.h: test-codegen.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+gdbus-test-codegen-generated.h: test-codegen.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.project. \
 		--generate-c-code gdbus-test-codegen-generated \
 		--c-generate-object-manager \
diff '--exclude=.git' -urN a/gio/tests/gdbus-object-manager-example/Makefile.am b/gio/tests/gdbus-object-manager-example/Makefile.am
--- a/gio/tests/gdbus-object-manager-example/Makefile.am	2016-10-22 01:17:59.000000000 -0400
+++ b/gio/tests/gdbus-object-manager-example/Makefile.am	2017-10-09 16:01:27.500410757 -0400
@@ -11,10 +11,8 @@
 	gdbus-example-objectmanager-generated-org.gtk.GDBus.Example.ObjectManager.Cat.xml	\
 	$(NULL)
 
-$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen
-	$(AM_V_GEN) UNINSTALLED_GLIB_SRCDIR=$(top_srcdir) \
-		UNINSTALLED_GLIB_BUILDDIR=$(top_builddir) \
-		$(PYTHON) $(top_builddir)/gio/gdbus-2.0/codegen/gdbus-codegen \
+$(GDBUS_GENERATED) : gdbus-example-objectmanager.xml Makefile
+	$(AM_V_GEN) gdbus-codegen \
 		--interface-prefix org.gtk.GDBus.Example.ObjectManager. \
 		--c-namespace Example \
 		--c-generate-object-manager \
