diff '--exclude=.git' -urN a/pambase.py b/pambase.py
--- a/pambase.py	2020-08-06 17:46:40.000000000 -0500
+++ b/pambase.py	2022-04-10 14:51:31.668155782 -0500
@@ -9,6 +9,7 @@
 	parser = argparse.ArgumentParser(description='basic Gentoo PAM configuration files')
 	parser.add_argument('--libcap', action="store_true", help='enable pam_caps.so module')
 	parser.add_argument('--passwdqc', action="store_true", help='enable pam_passwdqc.so module')
+	parser.add_argument('--consolekit', action="store_true", help='enable pam_ck_connector.so module')
 	parser.add_argument('--elogind', action="store_true", help='enable pam_elogind.so module')
 	parser.add_argument('--systemd', action="store_true", help='enable pam_systemd.so module')
 	parser.add_argument('--selinux', action="store_true", help='enable pam_selinux.so module')
diff '--exclude=.git' -urN a/templates/system-auth.tpl b/templates/system-auth.tpl
--- a/templates/system-auth.tpl	2020-08-06 17:46:40.000000000 -0500
+++ b/templates/system-auth.tpl	2022-04-10 14:48:21.280886737 -0500
@@ -39,6 +39,10 @@
 session		optional	pam_ssh.so
 {% endif %}
 
+{% if consolekit %}
+-session        optional        pam_ck_connector.so nox11
+{% endif %}
+
 {% if systemd %}
 -session        optional        pam_systemd.so
 {% endif %}
