diff --git a/Source/JavaScriptCore/b3/B3SwitchValue.cpp b/Source/JavaScriptCore/b3/B3SwitchValue.cpp
index 55a55e9d07584e3d34f5c7dd6deea9766ae04f6f..f8709b43865c9774f4a88a2305b505924f50ba1b 100644
--- a/Source/JavaScriptCore/b3/B3SwitchValue.cpp
+++ b/Source/JavaScriptCore/b3/B3SwitchValue.cpp
@@ -28,7 +28,7 @@
 
 #if ENABLE(B3_JIT)
 
-#include "B3BasicBlock.h"
+#include "B3BasicBlockInlines.h"
 #include <wtf/ListDump.h>
 
 namespace JSC { namespace B3 {
