From 377dd734242a07c720b3c3dd619240e0cece6ea8 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 8 Jun 2016 14:22:32 +0200
Subject: [PATCH 10/55] linux: Name the iDevice start poll timeout

---
 src/linux/up-device-idevice.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/linux/up-device-idevice.c b/src/linux/up-device-idevice.c
index 6ab1bf8..fc64c1f 100644
--- a/src/linux/up-device-idevice.c
+++ b/src/linux/up-device-idevice.c
@@ -69,7 +69,7 @@ up_device_idevice_poll_cb (UpDeviceIdevice *idevice)
 }
 
 static gboolean
-start_poll_cb (UpDeviceIdevice *idevice)
+up_device_idevice_start_poll_cb (UpDeviceIdevice *idevice)
 {
 	UpDevice *device = UP_DEVICE (idevice);
 	idevice_t dev = NULL;
@@ -160,7 +160,10 @@ up_device_idevice_coldplug (UpDevice *device)
 		      "has-history", TRUE,
 		      NULL);
 
-	idevice->priv->start_id = g_timeout_add_seconds (1, (GSourceFunc) start_poll_cb, idevice);
+	idevice->priv->start_id = g_timeout_add_seconds (1, (GSourceFunc) up_device_idevice_start_poll_cb,
+							 idevice);
+	g_source_set_name_by_id (idevice->priv->start_id,
+				 "[upower] up_device_idevice_start_poll_cb (linux)");
 
 	return TRUE;
 }
-- 
2.13.0

