From d5ec9d4f292726d1695f5154e546ac8536bf454d Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 23 Sep 2015 17:00:00 +0200
Subject: [PATCH 29/90] daemon: Print the filename when the config file is
 missing

---
 src/up-config.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/up-config.c b/src/up-config.c
index 703774a..2a17f30 100644
--- a/src/up-config.c
+++ b/src/up-config.c
@@ -114,13 +114,13 @@ up_config_init (UpConfig *config)
 					 G_KEY_FILE_NONE,
 					 &error);
 
-	g_free (filename);
-
 	if (!ret) {
-		g_warning ("failed to load config file: %s",
-			   error->message);
+		g_warning ("failed to load config file '%s': %s",
+			   filename, error->message);
 		g_error_free (error);
 	}
+
+	g_free (filename);
 }
 
 /**
-- 
2.13.0

