From c015e6b21e3cb8f5bc944564850d9ffc35a6a6c7 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 23 Sep 2015 16:14:40 +0200
Subject: [PATCH 28/90] linux: Fix deprecation warning in integration-test

---
 src/linux/integration-test | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/linux/integration-test b/src/linux/integration-test
index 48c0c1d..8388793 100755
--- a/src/linux/integration-test
+++ b/src/linux/integration-test
@@ -25,14 +25,17 @@ import unittest
 import time
 
 try:
+    import gi
     from gi.repository import GLib
     from gi.repository import Gio
+    gi.require_version('UPowerGlib', '1.0')
     from gi.repository import UPowerGlib
 except ImportError as e:
     sys.stderr.write('Skipping tests, PyGobject not available for Python 3, or missing GI typelibs: %s\n' % str(e))
     sys.exit(0)
 
 try:
+    gi.require_version('UMockdev', '1.0')
     from gi.repository import UMockdev
 except ImportError:
     sys.stderr.write('Skipping tests, umockdev not available (https://launchpad.net/umockdev/)\n')
-- 
2.13.0

